src/docfx/lib/JsonUtility.cs

```diff
@@ -468,18 +469,25 @@ protected override JsonProperty CreateProperty(MemberInfo member, MemberSerializ
 
                 void ShouldNotSerializeEmptyArray()
                 {
-                    if (s_rawContractResolver.ResolveContract(prop.PropertyType) is JsonArrayContract)
```

Comment:
minor: `s_rawContractResolver` was involved to avoid infinite recursion. Since we do not use `ResolveContract` here anymore, we can remove it and use `s_serializer.ContractResolver` instead?